go/vt/vtgate/vtgate.go

```diff
@@ -1065,6 +1072,13 @@ func recordAndAnnotateError(err error, statsKey []string, request map[string]int
 	request = truncateErrorStrings(request)
 
 	errorCounts.Add(fullKey, 1)
+	promstats.Add(promErrorCounts, map[string]string{
```

Comment:
Again in this case I think the Add should be called from the internals, either by always using CounterFunc from prom or by pushing the measures in using something like `WithLabelValues`:
https://github.com/prometheus/client_golang/blob/master/prometheus/counter.go#L171

Reply:
+1